dandi/metadata.py

```diff
@@ -603,6 +603,10 @@ def convertv1(data):
 
 def migrate2newschema(meta):
     newmeta = convertv1(meta)
+    if "version" in newmeta:
+        newmeta["id"] = f"{newmeta['identifier']}/{newmeta['version']}"
+    else:
+        newmeta["id"] = f"{newmeta['identifier']}/draft"
```

Comment:
`identifier` is made optional in `Identifiable` (and was/is optional for others such as `RelatedParticipant`) in this PR, so I guess it could fail or get a bogus `None` here?  may be no `id` should be assigned `if not identifier`?

Reply:
this is a converter and this function will go away as soon as we migrate.

Reply:
that is good, still don't see why it should be "not robust by design" meanwhile, but ok.  I will assign you an issue if I run into one ;-)